src/InteractiveWindow/Editor/History.cs

```diff
@@ -171,66 +176,84 @@ private Entry Move(string pattern, int step)
         {
             Debug.Assert(step == -1 || step == +1);
 
+            int end;
             bool wasLive = _live;
-            _live = true;
+            bool wasUninitialized = (_current == -1);
+            bool patternEmpty = string.IsNullOrWhiteSpace(pattern);
 
+            _live = true;
             if (Length == 0)
             {
                 return null;
             }
 
-            bool patternEmpty = string.IsNullOrWhiteSpace(pattern);
-
-            int start, end;
             if (step > 0)
             {
-                start = 0;
                 end = Length - 1;
-            }
-            else
-            {
-                start = Length - 1;
-                end = 0;
-            }
-
-            // no search in progress:
-            if (_current == -1)
-            {
-                _current = start;
-            }
-
-            int visited = 0;
-            while (true)
-            {
-                if (_current == end)
+                if (wasUninitialized) return null;
+             
+                while (true)
                 {
-                    if (visited < Length)
+                    if (_current == end)
```

Comment:
(Since it will be 1 in practice, feel free to assert that we actually saw equality.)